src/System.Diagnostics.Process/src/System/Diagnostics/Process.Linux.cs

```diff
@@ -83,6 +83,18 @@ public TimeSpan UserProcessorTime
             }
         }
 
+        partial void EnsureHandleCountPopulated()
+        {
+            if (_processInfo.HandleCount <= 0 && _haveProcessId)
+            {
+                string path = "/proc/" + _processId.ToString() + "/fd";
+                if (Directory.Exists(path))
```

Comment:
Fixed. I added a try/catch DirectoryNotFoundException to handle the case where the process is ended and the /proc/pid/ folder is cleaned up between the Exists and the GetFiles call.